### PR TITLE
Display appropriate GUI that accurately displays offline by design

### PR DESCRIPTION
**Pull request by Vlatombe**
Reworked computer icon and associated descriptions to be controlled by the `OfflineCause` implementation. For implementations such as `RetentionStrategy.Demand`, this allows to set an offline cause that can't be interpreted by the end user as an error.

* Allow the retention strategy to set an offline cause. For the on-demand retention strategy, this allows to provide a reason why the agent is offline, instead of leaving the agent offline with no message.
* Introduced `pause` and `computer-paused` symbols, used by the on-demand retention strategy to indicate the computer is offline on purpose and is not in error state
* Add `getOfflineCause` to `IComputer` and refactor related methods to pull up methods that make sense at the interface level.
* Extract `IOfflineCause` interface from `OfflineCause` to make it possible to render offline causes in other runtimes (CloudBees CI HA)

**Screenshots**

| Description | Before | After |
|--------|--------|--------|
| On-demand agent offline before any connection attempt | agent-ondemand-offline-before | agent-ondemand-offline |
| Agent overview | agent-set-overview-before | agent-set-overview |
| Agent temporary offline | agent-temporary-offline-user-before | agent-temporary-offline-user |
| Agent tooltip when offline due to idle | No tooltip | agent-tooltip-idle |
| Agent tooltip when temporary offline | No tooltip | agent-tooltip-temporary-offline | 

See [JENKINS-XXXXX](https://issues.jenkins.io/browse/JENKINS-XXXXX).

### Proposed changelog entries

- The agents online/offline status and icon can now be influenced by the offline cause, giving better information to users as well as clarifying some use cases when an agent was offline by design and not because of a configuration or technical error.

### Proposed upgrade guidelines

N/A

```[tasklist]
### Submitter checklist
- [X] The Jira issue, if it exists, is well-described.
- [X] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)). Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [X] There is automated testing or an explanation as to why this change has no tests.
- [X] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [ ] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [ ] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [ ] For new APIs and extension points, there is a link to at least one consumer.
```

### Desired reviewers

@mention

Before the changes are marked as `ready-for-merge`:

```[tasklist]
### Maintainer checklist
- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).
```

### CODE REVIEW

**Comment by mawinter69** [comment]
Could you add the new icon to https://github.com/jenkinsci/jenkins/blob/master/core/src/main/resources/hudson/model/ComputerSet/_legend.jelly
and probably update the not-accepting texts

**Comment by Vlatombe** [comment]
I have never even seen that "legend" popup. TIL

**Comment by Vlatombe** [comment]
> probably update the not-accepting texts

Actually not needed. The existing on-demand retention strategy was putting the agent offline, whereas the current scheduled retention strategy just marks it as suspended (so it actually is connected but can't be used).

File: core/src/main/resources/jenkins/agents/Messages_cs.properties
Comment (jglick) [inline core/src/main/resources/jenkins/agents/Messages_cs.properties:1]: I think you have some mojibake issues? Compare https://github.com/jenkinsci/jenkins/blob/3983ac69e2bf8972e1294e7c57714892e898cadf/core/src/main/resources/jenkins/agents/Messages_cs.properties#L1 to what I guess is the original, https://github.com/jenkinsci/jenkins/blob/3983ac69e2bf8972e1294e7c57714892e898cadf/core/src/main/resources/lib/hudson/executors_cs.properties#L29

Comment (Vlatombe) [reply core/src/main/resources/jenkins/agents/Messages_cs.properties:1]: Doublechecking...

File: core/src/main/java/hudson/slaves/OfflineCause.java
Comment (timja) [inline core/src/main/java/hudson/slaves/OfflineCause.java:223]: If this wasn't shown before and now is I'd suggest changing the text from `computer` to `agent`

`Offline because computer was idle` doesn't make sense as a computer isn't user facing terminology, (not mentioned in https://www.jenkins.io/doc/book/glossary/), it should be agent instead.

Comment (Vlatombe) [reply core/src/main/java/hudson/slaves/OfflineCause.java:223]: This was already shown before, just not before the agent was online, then became offline again as instructed by the retention strategy.

**Review by jglick — APPROVED** [review]

**Comment by timja** [comment]
can you add a changelog entry please

**Review by timja — APPROVED** [review]
/label ready-for-merge

----

This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

File: core/src/main/resources/hudson/slaves/OfflineCause/cause.jelly
Comment (daniel-beck) [inline core/src/main/resources/hudson/slaves/OfflineCause/cause.jelly:32]: Caused https://www.jenkins.io/security/issue/SECURITY-3669/

Perhaps something like this would benefit from `needs-security-review`?

**Pull request merged**
